api/apps/api/src/modules/projects/output-project-summaries/output-project-summaries.service.ts

```diff
@@ -0,0 +1,291 @@
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { AppConfig } from '@marxan-api/utils/config.utils';
+
+export const outputProjectSummaryFolder = 'output-project-summary/';
+export const outputProjectSummaryFilename = 'summary.csv';
+export const outputProjectSummaryScenariosFilename = 'scenarios.csv';
+
+@Injectable()
+export class OutputProjectSummariesService {
+  private readonly logger = new Logger(OutputProjectSummariesService.name);
+
+  constructor(
+    private readonly bestSolutionsDataService: BestSolutionDataService,
+    private readonly summedSolutionsDataService: SummedSolutionDataService,
+    @InjectRepository(Scenario)
+    private readonly scenarioRepo: Repository<Scenario>,
+    @InjectRepository(ProjectsPuEntity, DbConnections.geoprocessingDB)
+    private readonly projectsPU: Repository<ProjectsPuEntity>,
+    @InjectRepository(OutputProjectSummaryApiEntity)
+    private readonly outputSummaryRepo: Repository<OutputProjectSummaryApiEntity>,
+  ) {}
+
+  async saveSummaryForProjectOfScenario(scenarioId: string): Promise<void> {
+    const { projectId } = await this.scenarioRepo.findOneOrFail({
+      where: { id: scenarioId },
+    });
+
+    const planningUnits = (await this.projectsPU.find({ where: { projectId } }))
+      .map((pu) => pu.puid)
+      .sort();
+
+    const scenarios = await this.scenarioRepo.find({
+      select: { id: true, projectScenarioId: true, name: true },
+      where: { projectId },
+    });
+
+    const bestSolutionData = await this.getBestSolutionDataForScenarios(
+      scenarios,
+    );
+    const summedSolutionData = await this.getSummedSolutionDataForScenarios(
+      scenarios,
+    );
+
+    const workingDirectory = OutputProjectSummariesService.getProjectTempFolder(
+      projectId,
+    );
+    const csvPath = joinPath(workingDirectory, outputProjectSummaryFolder);
+    await this.createCSVFolder(csvPath);
+
+    const summaryCSVStream = this.arrangeSummaryCSV(
+      planningUnits,
+      scenarios,
+      bestSolutionData,
+      summedSolutionData,
+    );
+    const scenarioCSVStream = this.arrangeScenariosCSV(scenarios);
+
+    await this.writeCSV(
+      workingDirectory,
+      outputProjectSummaryFilename,
+      summaryCSVStream,
+    );
+    await this.writeCSV(
+      workingDirectory,
+      outputProjectSummaryScenariosFilename,
+      scenarioCSVStream,
+    );
+
+    const zipFullPath = this.zipCSVs(projectId, workingDirectory);
+
+    await this.saveSummary(projectId, zipFullPath);
+
+    await this.cleanupProjectTempFolder(projectId);
+  }
+
+  private arrangeSummaryCSV(
+    planningUnits: number[],
+    scenarios: Scenario[],
+    bestSolutionData: ProjectScenarioBestSolutionMapping,
+    summedSolutionData: ProjectScenarioSummedSolutionMapping,
+  ): CsvFormatterStream<FormatterRow, FormatterRow> {
+    const projectScenarioIds = scenarios
+      .map((scenario) => scenario.projectScenarioId)
+      .sort();
```

Comment:
I assumed that having the type number[] it would apply a sorting algorithm that makes sense for that, but yeah, you're right; this will fail for larger planningUnitIds

Reply:
yep, this being Yavascript, WAT is always around the corner: https://www.destroyallsoftware.com/talks/wat

to be fair, once the TS is transpiled to JS, types are erased so the `Array.prototype.sort()` function is necessarily (?) going to be dumber than we'd wish 🤷🏼 

another option would be to use _true_ JS typed arrays such as `UInt32Array`, for which the `.sort()` method would do a proper numeric sorting, but your fix is idiomatic and doesn't rely on creating new typed arrays out of primitive arrays.